### DISCUSSION TITLE
New dark/light mode feedback

### DISCUSSION THREAD

**Discussion by mifi** [mifi/lossless-cut#1507] [Ideas]
With [v3.53.0](https://github.com/mifi/lossless-cut/releases/edit/v3.53.0) comes a redesign of the UI colors to make it more consistent with most other dark mode apps these days. It also includes a new "light mode"

Feedback is most welcome!

[leave a vote!](https://twitter.com/losslesscut/status/1634381935232561152)

![Screenshot 2023-03-10 at 23 13 35](https://user-images.githubusercontent.com/402547/224352613-4d8cbc2e-f68e-4026-86fb-19aa85af68e1.jpg)
![Screenshot 2023-03-10 at 23 13 53](https://user-images.githubusercontent.com/402547/224352621-517c8322-3866-46e0-b591-f43b6995b790.jpg)

**Comment by Dean-Corso** [comment]
Hi @mifi,

why do you guys care so much about adding a dark mode theme? I really hate that and the light theme too because it's too light! What about the default theme? So if you want you can add another themes the user can choose but just don't remove the damn default theme! There are also apps out there who just using dark theme only (not in older versions) what is total dumb! One more time, you can add themes but do not remove the gray colored default theme you were using before, so I like that one. Thank you.

**Comment by mifi** [comment]
I’ve had a lot of requests about wanting a proper dark experience, and it seems to be the norm in most apps now, that’s why I added it. *I don’t want to support themes as it’s too much of a maintainence burden*. Dark/light colors is based off of Radix UI colors design:
https://www.radix-ui.com/docs/colors/palette-composition/understanding-the-scale
Meaning I don’t have to implement any themes, I just choose colors from their scales, and dark/light mode «just works».
However if it turns out most people don’t like the dark mode, I can go back to having just 1 theme.

What is it exactly that you *hate* about the dark scheme? Any constructive feedback? I believe benefits include:
- More contrast
- Easier to read text
- Easier to distinguish segment colors

**Comment by Leejigwei** [comment]
It's not working since version 3.52, how can I fix it?
window 7 X64 system
![捕获](https://user-images.githubusercontent.com/127607577/224488740-8728ce9f-fa02-41a8-9c3b-5a2a4acb13e1.PNG)

**Reply by ld892012** [reply]
Windows 7 is not supported after v3.50
[https://github.com/mifi/lossless-cut/discussions/1476](https://github.com/mifi/lossless-cut/discussions/1476)

**Reply by Leejigwei** [reply]
Ok thanks

**Comment by ld892012** [comment]
I think no new features is bad as long as the user can choose to use it or not, personally i like dark theme at night and light theme at day :+1: 

Maybe give possibility to use dark/light theme or just default theme ?
And maybe add an option to makes LosslessCut follows color theme of the OS ?
I have some difficulties to see keyframes in segment when in dark theme, maybe darken keyframes in dark theme ?

Anyway this is a cool feature and a great software ^^

**Reply by sempiternoiddqd** [reply]
i'm in 3.61 there's not light mode only dark, do you know how to change it?

**Reply by sujfuy** [reply]
Mine was dark, it switched to light by itself and I cannot find any way to change it or any instructions on how to do it.

**Reply by mifi** [reply]
Hmm! If you click the "toggle advanced view" button and then the  sun icon, nothing happens? 
<img width="25" alt="Screenshot 2024-06-06 at 12 24 31" src="https://github.com/mifi/lossless-cut/assets/402547/448fba2d-a27e-48f8-a29f-a3e2b2a8bd34">

**Comment by Dean-Corso** [comment]
@mifi,

> What is it exactly that you hate about the dark scheme?

The problem is the colors itself. They are just not pleasant for my eyes and I need to have some more sensitive colors, no aggressive ones, to see everything better. In your case, the dark & light themes are just too intensive, and I really hate developers who just force users to use any specific theme or keyboard shortcuts only etc. They never think about people who having some handicaps. Normally you can add any style themes / owner draws you want but just keep one default basic theme with decent colors or just using Windows colors etc. Adding custom style themes should be only done as an optional feature, without to force them. I see many apps who do force users with any dark themes what is really a NO-GO, but they just don't care and do not think forward in more than just one direction and that's the problem which you can all limited thinking. I also did complain that color issue in Chrome / Brave browser too when using the private mode, so there you can not use any own template themes anymore (only in not private mode). They just removed that feature on purpose without any logic behind that step, what means we users getting forced into using dark theme (Chrome) and purple theme (Brave) only in private mode (my default mode). I told them about that almost a year ago (on GitHub) but they just don't care and that makes me furious! All in all it is your choice whether you just add that dark & light theme or go back to your single default theme, what I do prefer. Maybe not easy for you to make a decision if you can not add all three themes into your app. Anyway, so it was just my opinion about those specific colored themes etc.

**Reply by mifi** [reply]
You’re saying the dark mode colors are too intense. Do you then mean the colored segments, export button etc? Those are the only things that have actual chromatic colors, the rest is black and shades of gray. If so, should the segment colors be dimmed or desaturated?

or do you mean the contrasts are too high? E.g the white text is too strong against the black background.

or is it that you don’t like the color black and prefer gray instead?

or something else?

I understand that you have alot of hate towards certain developers of software that push color themes on you, but LosslessCut is open source so you are not forced to do anything. You can stay on an older version, or you can change the colors yourself in the code. (Find-replace all `var(—gray1)` wirh #777 and you have a gray background again.
I’ll try to listen to users’ opinions, but if it’s just one or two people who don’t like dark mode, and 10.000 others like it, then it’s gonna be hard to justify removing it. I think that’s probably the resoning for other software too, if they «don’t listen to you», is because they cater to the masses.

**Comment by tcz305** [comment]
The progress bar on the white theme is awful ....don't know if it is finished yet but you can't see anything with all these multicolors ....light brown on white etc. Maybe a more neutral palette than all the bold colors would be easier on the eyes.
The dark theme like someone else said, is too dark although it's not that bad, some people have issue with light text on very dark backgrounds ....maybe if you can make it closer to #424242 in this image palette 
<a href='https://postimg.cc/SJMFHFWt' target='_blank'><img src='https://i.postimg.cc/SJMFHFWt/style-color-themes-dark1-1190560670.png' border='0' alt='style-color-themes-dark1-1190560670'/></a>
But I think the present color would be fine if you made all the colors a bit more neutral rather than bold.
Thanks

*ok looking at it a bit more, the dark theme as is is fine, it is the progress bar which is too dark with bold colors on top.
The white theme is awful. If you are going to use a white theme, there should not be multicolors all over the place .....neutral light blues and greys and other light shades.
(and the progress bar should be darker in the white theme ....it dissapears into the background now.)

**Reply by mifi** [reply]
@tcz305 thanks for your feedback. Just to be sure, what you mean by "progress bar" is the segment timeline, right?

<img width="1115" alt="Screenshot 2023-03-12 at 17 13 36" src="https://user-images.githubusercontent.com/402547/224535434-0ceb1cf5-18a3-4d85-8c98-595d45cd1732.png">

you think the segment colors are too strong in both dark mode and light mode? You think they should be more desaturated?

maybe something like this:

<img width="1115" alt="Screenshot 2023-03-12 at 17 27 20" src="https://user-images.githubusercontent.com/402547/224536045-d16ac62f-c8c8-4747-b0ea-d229b252aaaf.png">

<img width="1118" alt="Screenshot 2023-03-12 at 17 28 09" src="https://user-images.githubusercontent.com/402547/224536060-f8584777-59a4-413e-830c-16100890f2a8.png">

**Comment by tcz305** [comment]
yes, that is a lot better, but the progress bars themselves need to be differentiated from the rest of the ui ...

something like this (excuse the crude renderings)

![224536045-d16ac62f-c8c8-4747-b0ea-d229b252aaaf b](https://user-images.githubusercontent.com/19213465/224561863-4765d49a-7476-4eee-9f89-cddb8b06507e.jpg)
![224536060-f8584777-59a4-413e-830c-16100890f2a8 b](https://user-images.githubusercontent.com/19213465/224561865-3cc7d7f0-7b96-4475-b7a6-88331546ef08.jpg)

also the buttons would look better with more neutral colors, something like this

![SCR-20230312-isq b](https://user-images.githubusercontent.com/19213465/224562245-cc792ca2-9632-44bb-a0f7-617acc06a698.jpg)

![SCR-20230312-izt b](https://user-images.githubusercontent.com/19213465/224562496-b3a23bb0-e6dd-4823-a5d7-cf939298e3b2.jpg)

**Comment by Dean-Corso** [comment]
Hi again,

the main area color which you changed from gray to dark is too dark and the light color too light or white if you want. You need to use some neutral color values which are not so intensive. Even Chrome browser is using brighter dark colors as you do in LC. The user @tcz305 did post a color image of gray 800 which would be a better choice for the entire app background (most area). Also, using a total black colored title bar is also no good idea. If you want to use dark theme colors then have a look at Photoshop CS6 which was using a grayed soft color background (title bar / tools etc.). I made an image where you can see the main color of Photoshop, which is better.
![PS_2023-03-13_202221](https://user-images.githubusercontent.com/88513086/224812766-d39a059b-6b3e-43d7-bdfd-c64e0dcad7af.png)
Maybe you could try this brighter gray color value of PS for your dark mode background & title bar. As I said before, if developer of apps wants to use specific themes like that dark one then they should not only use this one as only theme. A normal default theme without any specific set colors (default used Windows themed colors by user) should keep in the app too. I really don't understand the entire hype about that dark color theme which was started by Adobe in PS if I remember right and since that time many devs want to implement that dark theme too for their own apps. Ok alright so if they want they can do that, but please do not just use this one as only theme in the app. I have problems with dark backgrounds at all and prefer neutral colors. Also, websites using often just dark themes only what makes me angry too of course. About the light theme colors in your app, which are also too light. What about using Sepia color for the light theme? So this color I'm using as background for reading text in Firefox reading view, which is very gentle.

So if there are 10 TSD users who like the dark theme and just 10 who do not, then you would think you must use the theme the most do like right. But just remember, before you were using the dark & light theme nobody did complain it as it was not implemented or? Anyway, maybe you could just adjust the colors a little more (not too dark & light) and make them more gentle, or maybe you could add a new option to set those colors values plus and minus manually a little or so.

**Comment by Mr-Hazem-Eng-Artist** [comment]
I really like the new Dark UI you made, yeah I know somepeople have issues saying it is (Too dark) but also the degree of grey they want is a bit too light to be dark ,, if you want my honest advice? make 3 UIs one of them is the same degree of light grey of your classic UI & the 2 others are your new Dark & Lights ones, OR ,  Do like Photoshop (it has 4 color degrees that suits everybody I guess)
![image](https://user-images.githubusercontent.com/81746929/227551527-8b65daaa-56e0-48b3-bcd6-deadc4861ce8.png)

Just two things, this new UI is slowing down losslessCut abit, your past version 3.52 opens on my PC in 1 second, current 3.53 with the new UI opens in 3 or 4 seconds, I am on WIndows 11 22H2 March Update with i9 12900k & RTX 3090 .. also the title bar menus are laggy, open (File) menu & drag the mouse till you reach (window) title until the menu disappears & new menu appears instead.

**Reply by mifi** [reply]
I think having multiple color schemes is gonna be a bit too much maintenance, but I plan on trying to find a better middle-way

**Reply by mifi** [reply]
About the startup slowness, I don't think it has anything to do with the UI changes. I think it could be related to one of these changes:
- d252ed3a145b8f12994e2a0b3172dfb270822e9d
- 86f451539a28464cedf16f0b0abf787a2b9b56fd

However for me on MacOS it's no slower than 3.52, so I'm not sure what's happening there.

As for the menu bar slowness, I don't experience that on MacOS. Is that better in 3.52 also? could you record a screen cast?

**Reply by Mr-Hazem-Eng-Artist** [reply]
I know it is not easy to make available several UI options, but you know, you can't make everybody satisfied anyways ;) . Regarding the slowness, I recorded 40 seconds video (only 3 MB) take a look at it here when you can. In general throughout Windows 10 & 11, version 3.53 is the most sluggish to me , I use losslesscut since almost 2 years.
https://drive.google.com/file/d/1Cjt-qpQFrHVhN-_cQOrjf7n9sgatlvhx/view?usp=sharing

**Reply by mifi** [reply]
thanks for recording. is it really just 1 second from double click to "DROP FILES" visible on 3.52? Also the menu slowness issue is not present on 3.52?

**Reply by Mr-Hazem-Eng-Artist** [reply]
Version 3.52 was a bit faster, but that slow/sluggish menu issue is also existed in version 3.52 & not in 3.53 alone, if it is something you can work on, then do it, if not, then just concentrate on your schedule about the new UI color and others. After all, v 3.53 is still usable & it is not ruining the cut/edit experience :)

**Reply by mifi** [reply]
Actually I could work on it, but only if I knew which version the menu started to become slow. And if you could show me a recording of startup in v 3.52 showing how it looks like when it boots up in just 1 sec. Because the flashing background colors gives me hints as to where the slowdown happens.

**Reply by Mr-Hazem-Eng-Artist** [reply]
OK I tried something, I still have copy of v3.52 (zip portable version, also my 3.53 was portable) it is the very exact same like v3.53 about the startup speed & the sluggish menus! The difference is that now I am using a fresh Windows 11 installation, before that, I had 3.52 on Windows 10 ! if that is the case then at least we know that losslesscut is working but might be a bit not as responsive on Windows 11 as it is on WIndows 10 ?

**Reply by mifi** [reply]
That’s still not very comforting. As windows 11 is the future and should be fast. Anyways if you can record screencasts showing a definite difference between two versions of losslesscut then i can look into it. If not then i dont know how to fix it because i dont run windows myself

**Reply by Mr-Hazem-Eng-Artist** [reply]
For the moment I can't record a video for Losslesscut on Windows 10 as I only have Windows 11 instead. Anyways if I later install Windows 10 again here or there I will try to record a video. For now you can concentrate on the UI colors & any bug fixes you wanna include in the next build & just keep the (current speed of lossless cut that you saw in the past recorded video) as is, it is still acceptable, I know other editing programs that are way slower than that , making losslesscut look like a fast rabbit.

**Comment by Aespaaa** [comment]
Hey just wondering if there is any way to change the color for the orange segment buttons at the bottom? 
![Screenshot (2430)](https://user-images.githubusercontent.com/128879543/227693793-bc64acfe-9fe5-4637-8ce8-34ddba4f4b5d.png)

**Reply by mifi** [reply]
those colors follow the color of the current segment, but i plan on toning them down a bit, like this:

<img width="934" alt="Screenshot 2023-03-25 at 23 38 48" src="https://user-images.githubusercontent.com/402547/227723953-84d31ac0-2a16-4803-b685-ac8c3c4fd1df.png">

which color did you have in mind?

**Reply by Aespaaa** [reply]
I was just thinking a darkish blue like the play button , something like this 
![Screenshot](https://user-images.githubusercontent.com/128879543/227764826-9bfe29e1-94a9-402c-894d-de4b8f7c5854.jpg)

 it would always be nice to have the option to choose any color i would like , maybe a color wheel of some sort since not everyone's preference would be the same , thanks anyways!

**Comment by SanderBouwhuis** [comment]
I'm using Win10, and the dark theme looks fine to me. Also, the menus are fast.
Finally, I tried to do some cuts and they too are fine.

Big thumbs up on the dark theme!

**Comment by mifi** [comment]
Have been working on neutralising colors more as well as adding more borders to make UI elements stand out more:

<img width="1080" alt="Screenshot 2023-04-02 at 23 56 56" src="https://user-images.githubusercontent.com/402547/229360955-39f5e8d0-e31a-416d-ac2b-7678eef355a7.png">

<img width="1080" alt="Screenshot 2023-04-02 at 23 56 51" src="https://user-images.githubusercontent.com/402547/229360981-6316ab8f-bb4e-44d5-8884-082f4d54ee78.png">

**Comment by SanderBouwhuis** [comment]
A bit of a shame you are making the dark theme look pale and sad. I love the current bright and happy colours.

**Reply by mifi** [reply]
I was thinking about maybe making an option for color strength, as it seems to be the most polarizing.

**Comment by JCernyynreCJ** [comment]
I also prefer the old grey theme. The problem isn't that the new dark/light themes were added, but the fact that the old grey one was removed.

**Reply by mifi** [reply]
can't please everyone 🤷 You can still use the old version though

**Reply by JCernyynreCJ** [reply]
Yes, I get that, but adding the new themes while keeping the old one, i.e. having 3 themes, would solve this entire issue. The color itself isn't the only thing. The old grey theme had the top buttons highlighted and, in general, everything was more visible and easier to find/click on.

**Reply by mifi** [reply]
We can probably improvethe button situation by making them stand out more.
As for having three themes, it would increase the maintainence burden and cost to develop new features when everything has to be tested 3 times. At least now when using radix ui’s color values, I just design it using dark mode,and most things «just work» in light mode too, so i dont need to test too much. Also the light mode is more like a «bonus» add-on. I consider the dark UI to be the main Ui.

But of course if someone has any ideas for css based UI frameworks that handle multiple colors/themes in a way that it supports multiple themes while requiring no additional work for  developers and designers, I could be open for swapping out radix ui’s css color framework.

**Reply by JCernyynreCJ** [reply]
Perhaps make the dark theme a bit lighter then? Other prominent editors like Gimp, Shotcut, PyCharm etc. use very similar shade of grey as default. And indeed the shade they use looks to me (and many others, as the programs have been using the same default theme for years) to be gentler to eyes and more comfortable to watch for longer time.

![Gimp-Shotcut-PyCharm](https://user-images.githubusercontent.com/100674469/230732858-1c0de4af-ceee-4c1a-ad8c-9ac275e3e820.jpg)

**Reply by mifi** [reply]
Actually I would say LosslessCut has the same shade of gray as shotcut has (lighter than PyCharm):

<img width="454" alt="Screenshot 2023-04-09 at 11 47 05" src="https://user-images.githubusercontent.com/402547/230751468-98996caf-da4a-4ccc-b9bc-dca25a948ea3.png">

Yes GIMP is lighter, but that's only one example.

I'm using the radix color index 4 from https://www.radix-ui.com/docs/colors/palette-composition/understanding-the-scale for the UI background. It's called `Hovered UI element background`. So actually it's already lighter than the color I'm supposed to use which is `3: UI element background`. If I use lighter colors, it's gonna clash with other colors like:
- 5: Active / Selected UI element background
- 6: Subtle borders and separators
- 7: UI element border and focus rings

...etc

**Reply by JCernyynreCJ** [reply]
When compared directly, LosslessCut's dark theme is noticeably darker than PyCharm. Especially LLC's outmost border/window looks almost solid black. As stated by others above, the dark/light themes are too extreme. A middleground, if possible, would be thus welcomed. 

![PyCharm-LosslessCut](https://user-images.githubusercontent.com/100674469/230756457-d17c046b-68fd-480d-bb32-0f19535b36bb.jpg)

There is also an issue with the track selection window. In the old version, when exporting a video and clicking on the tracks option ("Keep X tracks") in the export menu, the track menu opens over the export menu and can be used to select/deselect tracks just before exporting. In the current version, the new display-wide track menu opens under the export menu and cannot be interacted with.

**Reply by mifi** [reply]
While I agree that it's darker than  PyCharm, I believe losslesscut has the same shade of gray as shotcut, so it shouldn't be that controversial, unless shotcut's colors are also considered too dark by many people.

> There is also an issue with the track selection window. In the old version, when exporting a video and clicking on the tracks option ("Keep X tracks") in the export menu, the track menu opens over the export menu and can be used to select/deselect tracks just before exporting. In the current version, the new display-wide track menu opens under the export menu and cannot be interacted with.

See #1540

**Comment by Mr-Hazem-Eng-Artist** [comment]
An Update regarding the UI of version 3.54 , now it opens very fast just like it was with me on Windows 10, it opens in just 1 second, this happened after Windows 11 update which is (KB5023778 updates Windows 22H2 to build number 22621.1485) , So, yeah the faster more responsive start of Losslesscut is finally available on Win11 that Update I mentioned, below here is a video I recorded for that one second start, but sadly the sluggish Top Menus are still there but i can deal with it, & for the Dark UI, I still like that Dark look of version 3.54 , Speaking of myself, I know others fell it is more darker than it should & I don't blame them as everyone has their own taste, but personally I like the current UI :
(https://drive.google.com/file/d/13vlEjMH8bs41Mf-cepZwCffgX97As8wZ/view?usp=sharing)

**Comment by sempiternoiddqd** [comment]
Hello guys, i'm using 3.61.1 on win 10 and there's no option to change darkmode. Does it still have a light mode?

edit:
workaround: ~appdata/roaming/losslesscut/config.json 

"darkMode": false,

**Reply by mifi** [reply]
Hmm! If you click the "toggle advanced view" button and then the  sun icon, nothing happens? 
<img width="25" alt="Screenshot 2024-06-06 at 12 24 31" src="https://github.com/mifi/lossless-cut/assets/402547/448fba2d-a27e-48f8-a29f-a3e2b2a8bd34">

strange cause it works for me
